app/scripts/metamask-controller.js

```diff
@@ -4177,50 +4183,73 @@ export default class MetamaskController extends EventEmitter {
   // Hardware
   //
 
-  async getKeyringForDevice(deviceName, hdPath = null) {
+  /**
+   * Select a hardware wallet device and execute a
+   * callback with the keyring for that device.
+   *
+   * Note that KeyringController state is not updated before
+   * the end of the callback execution, and calls to KeyringController
+   * methods within the callback can lead to deadlocks.
+   *
+   * @param {object} deviceSelector - The device selector
```

Comment:
Nit: Perhaps we can organize these parameters differently. `deviceSelector` implies that the values are used to select the keyring, but that is not true of `hdPath`. It looks like `hdPath` is set on the keyring that we find, it isn't used for selecting at all.

Maybe we can give this a more generic name like `options` so that we can specify that the `name` is used for selection, but the `hdPath` is set on whatever keyring we find. Or maybe we could remove the parameter completely, and set the `hdPath` inside the callback in the three places we use this parameter.

Reply:
I agree, `options` makes more sense! Changed in [ee14fba](https://github.com/MetaMask/metamask-extension/pull/25435/commits/ee14fbab78cd0e53f7c278ba4af2dae25f785444)